internal/providers/terraform/azure/event_hubs_namespace.go

```diff
@@ -0,0 +1,164 @@
+		}
+	}
+	if d.Get("dedicated_cluster_id").Type != gjson.Null {
+		sku = "Dedicated"
+		meterName = "Capacity Unit"
+	}
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if u != nil && u.Get("monthly_ingress_events").Type != gjson.Null {
+		events = decimalPtr(decimal.NewFromInt(u.Get("monthly_ingress_events").Int()))
+	}
+
+	costComponents = append(costComponents, eventHubsThroughPutCostComponent(region, sku, meterName, capacityUnit))
+
+	if strings.ToLower(sku) == "basic" {
+		costComponents = append(costComponents, eventHubsCostComponent(region, sku, events))
+	}
+	if strings.ToLower(sku) == "standard" {
+		costComponents = append(costComponents, eventHubsCostComponent(region, sku, events))
+		costComponents = append(costComponents, eventHubsCaptureCostComponent(region, sku))
+	}
+
+	if strings.ToLower(sku) == "dedicated" {
+
+		if u != nil && u.Get("extended_retention_storage_gb").Type != gjson.Null {
+			retentionGB := []int{1000}
+			retention = decimalPtr(decimal.NewFromInt(u.Get("extended_retention_storage_gb").Int()))
+			retentionQuantites := usage.CalculateTierBuckets(*retention, retentionGB)
+			if retentionQuantites[1].GreaterThanOrEqual(decimal.NewFromInt(1)) {
+				costComponents = append(costComponents, eventHubsExtensionRetentionCostComponent(region, sku, &retentionQuantites[1]))
+			}
+		} else {
+			costComponents = append(costComponents, eventHubsExtensionRetentionCostComponent(region, sku, retention))
+		}
+
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func eventHubsCostComponent(region, sku string, quantity *decimal.Decimal) *schema.CostComponent {
+	if quantity != nil {
+		quantity = decimalPtr(quantity.Div(decimal.NewFromInt(int64(1000000))))
+	}
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("Ingress event (%s)", sku),
+		Unit:            "1M events",
+		UnitMultiplier:  1,
+		MonthlyQuantity: quantity,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("azure"),
+			Region:     strPtr(region),
+			Service:    strPtr("Event Hubs"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", ValueRegex: strPtr(fmt.Sprintf("/%s Ingress Events/i", sku))},
+			},
+		},
+		PriceFilter: &schema.PriceFilter{
+			PurchaseOption: strPtr("Consumption"),
+		},
+	}
+}
+
+func eventHubsThroughPutCostComponent(region, sku, meterName string, capacityUnit *decimal.Decimal) *schema.CostComponent {
+	meterName = fmt.Sprintf("%s %s", sku, meterName)
+	return &schema.CostComponent{
+		Name:            "Throughput",
+		Unit:            "units",
+		UnitMultiplier:  1,
+		MonthlyQuantity: capacityUnit,
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("azure"),
+			Region:     strPtr(region),
+			Service:    strPtr("Event Hubs"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "skuName", ValueRegex: strPtr(fmt.Sprintf("%s/i", sku))},
+				{Key: "meterName", Value: strPtr(meterName)},
```

Comment:
Here's what I checked for this:
```
db.products.distinct("attributes.meterName", {"vendorName": "azure", "service":"Event Hubs"});
[
	"Basic Ingress Events",
	"Basic Throughput Unit",
	"Capture",
	"Dedicated Capacity Unit",
	"Extended Retention",
	"Kafka Endpoint",
	"Processing Unit",
	"Standard Ingress Events",
	"Standard Throughput Unit"
]
```

1. Please check that this is using the correct filters, I guess `Dedicated Capacity Unit` is the one we want to use here. 

2. `Processing Unit` is for the Premium tier, but can that even be provisioned in Terraform? I only see Standard and Basic SKUs in https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/eventhub_namespace and we use Dedicated if the dedicated_cluster_id is set.

Reply:
yes we are using the Dedicated Capacity Unit, yes we cannot provision Premium